fnet/src/vespa/fnet/packetqueue.cpp

```diff
@@ -27,17 +27,27 @@ FNET_PacketQueue_NoLock::ExpandBuf(uint32_t needentries)
 
         uint32_t rOfs = _out_pos;
         uint32_t rLen = (_in_pos - _out_pos);
+//TODO Rewrite to pure C++
+#pragma GCC diagnostic push
+#if __GNUC__ >= 8
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
+#endif
         memcpy(newbuf + rOfs, _buf + rOfs, rLen * sizeof(_QElem));
+#pragma GCC diagnostic pop
     } else {                      // WRAPPED
         // BUFFER: |....................|
         // USED:   |######........######|
         //          r1Len          r2Len
 
         uint32_t r1Len = _in_pos;
         uint32_t r2Len = (oldsize - _out_pos);
+#pragma GCC diagnostic push
+#if __GNUC__ >= 8
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
+#endif
         memcpy(newbuf, _buf, r1Len * sizeof(_QElem));
-        memcpy(newbuf + _bufsize - r2Len, _buf + oldsize - r2Len,
-               r2Len * sizeof(_QElem));
+        memcpy(newbuf + _bufsize - r2Len, _buf + oldsize - r2Len, r2Len * sizeof(_QElem));
```

Comment:
It should also work to `static_cast<void*>` the destination parameter, but I actually prefer having the in-your-face pragmas since they don't hide the fact that something smells and should be fixed.

Reply:
Yes, dirty tricks could probably fix this.
I just wanted to annotate it clearly so @havardpe could figure out how to solve it in a more c++'ish way.